### PR TITLE
Add repo dump feature

### PR DESCRIPTION
**Pull request by jku**
* This adds a way to manually dump repository metadata
* the action enables dumping and uploads the repository dump as an artifact in the GitHub workflow run
* There is also and example added in the README and some cleanup commits

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,13 +52,96 @@ make test-python-tuf
...
+we can now run the test:
+```bash
+./env/bin/pytest tuf_conformance \
+    -k test_targets_version                 # run a specific test only
+    --entrypoint "./clients/go-tuf/go-tuf"  # use the included go-tuf client as client-under-test
+    --repository-dump-dir /tmp/test-repos   # dump repository contents
+
+# take a look at the repository targets.json that got debug dumped
+cat /tmp/test-repos/test_targets_version/1/targets.json
+cat /tmp/test-repos/test_targets_version/1/snapshot.json
```
Comment (jku) [inline README.md:90]: Maybe the repository version numbers should mention "refresh" so it's easier to figure out that a version is stored per client refresh... so `/tmp/test-repos/test_targets_version/refresh-1/targets.json` in this example

**Comment by AdamKorcz** [comment]
@jku LGTM. I think it is good to merge after fixing the conflict in `tuf_conformance/client_runner.py`

**Comment by jku** [comment]
rebased on main.

The workflow run page now has artifacts for each test suite run that contains the repository versions for each test: https://github.com/theupdateframework/tuf-conformance/actions/runs/10196881358?pr=90 (and the debugging workflow documented in the README should work)

**Comment by AdamKorcz** [comment]
Sounds great. Still a minor conflict, probably from the recent PR I merged after you rebased.

**Comment by jku** [comment]
another rebase on main

**Review by AdamKorcz — APPROVED** [review]

**Pull request merged**